src/neighborhoods.jl

```diff
@@ -9,6 +9,13 @@ A neighborhood of points in a spatial object.
 """
 abstract type AbstractNeighborhood end
 
+"""
+    AbstractBallNeighborhood
+
+A ball-like neighborhood of points in a spatial object.
+"""
+abstract type AbstractBallNeighborhood <: AbstractNeighborhood end
```

Comment:
Can we just keep the original `AbstractNeighborhood` type? Why we need an extra level of abstraction? Are you doing because of dispatch issues somewhere else?

Reply:
I added the AbstractBallNeighborhood because there is a NeighborhoodSearch and a search method specific for Union(BallNeighborhood, EllipsoidNeighborhood)

Reply:
I don't know if I understood the comment. I thought this PR is adding the `EllipsoidNeighborhood` right? How there is an existing method with it? If you can clarify why we need an extra abstract type, that would be helpful 👍🏽

Reply:
"method" might not be the correct term. Putting in other words, the `EllipsoidNeighborhood` is a `BallNeighborhood` with Mahalanobis metric as we discussed initially. The `AbstractBallNeighborhood` was defined to also consider Ellipsoid into two Ball specific calls inside `src/neighborsearch/neighborhood.jl`. Check the changes in that file, it might get clearer why I made it

Reply:
I will look into it more carefully later. I think I got your point. You are making a common parent type so that the KD tree search is triggered for both BallNeighborhood and EllipoidNeighborhood right? Since these two are types of "Balls". I wonder if we should treat them completely separate cases now that the EllipsoidNeighborhood is a separate struct. Perhaps avoiding the common parent AbstractBallNeighborhood and just using AbstractNeighborhood? That would just need a new entry in  the `Union{}`of the appropriate method I guess. I will think more about it too.